pep-0683.rst

```diff
@@ -245,9 +272,107 @@ would be noticed.
 Again, the only realistic way that the magic refcount would be reached
 (and then reversed) is if it were done deliberately.  (Of course, the
 same thing could be done efficiently using ``Py_SET_REFCNT()`` though
-that would be even less of an accident.) At that point we don't
+that would be even less of an accident.)  At that point we don't
 consider it a concern of this proposal.
 
+On 32-bit builds it isn't so obvious.  The magic refcount would be 2^28.
+Using the same specs as above, it would take roughly 1 second to
+accidentally immortalize an object.  Under reasonable conditions, it
+is still highly unlikely that an object be accidentally immortalized.
+It would have to meet these criteria:
+
+* targeting a non-immortal object (so not one of the high-use builtins)
+* the extension increfs without a corresponding decref
+  (e.g. returns from a function or method)
+* no other code decrefs the object in the meantime
+
+Under those conditions it would reach accidental immortality (on 32-bit)
+in, at most, a year if it averaged at least one of those increfs every
+158 seconds on that hypothetical workstation.  Of course, then it would
+have to run through the same number of (now noop-ing) decrefs before
+that one object would be effectively leaking.  This is highly unlikely,
+especially because we assume no decrefs.
+
+Furthermore, this isn't all that different from how such 32-bit extensions
+can already incref an object past 2^31 and turn the refcount negative.
+If that were an actual problem then we would have heard about it.
+
+Between all of the above cases, the proposal doesn't consider
+accidental immortality a problem.
+
+Stable ABI
+''''''''''
+
+The implementation approach described in this PEP is compatible
+with extensions compiled to the stable ABI (with the exception
+of `Accidental Immortality`_ and `Accidental De-Immortalizing`_).
+Due to the nature of the stable ABI, unfortunately, such extensions
+use versions of ``Py_INCREF()``, etc. that directly modify the object's
+``ob_refcnt`` field.  This will invalidate all the performance benefits
+of immortal objects.
+
+However, we do ensure that immortal objects (mostly) stay immortal
+in that situation.  We set the initial refcount of immortal objects to
+a value high above the magic refcount value, but one that still matches
+the high bit.  Thus we can still identify such objects as immortal.
+(See `_Py_IMMORTAL_REFCNT`_.)  At worst, objects in that situation
+would feel the effects described in the `Motivation`_ section.
+Even then the overall impact is unlikely to be significant.
+
+Accidental De-Immortalizing
+'''''''''''''''''''''''''''
+
+32-bit builds of older stable ABI extensions can take `Accidental Immortality`_
+to the next level.
+
+Hypothetically, such an extension could incref an object to value on
```

Comment:
This doesn't read well, maybe:
```suggestion
Hypothetically, such an extension could incref an object to a value on
```
But then the next line still doesn't parse for me.